src/iam/permhandler/permhandler.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/permhandler.hpp"
+#include "log.hpp"
+
+namespace aos {
+namespace iam {
+namespace permhandler {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+RetWithError<StaticString<uuid::cUUIDStrLen>> PermHandler::RegisterInstance(
+    const InstanceIdent& instanceIdent, const Array<FunctionalServicePermissions>& instancePermissions)
+{
+    LockGuard lock(mMutex);
+
+    LOG_DBG() << "Register instance: instance = " << instanceIdent;
+
+    Error                           err;
+    StaticString<uuid::cUUIDStrLen> uuidSecret;
+
+    Tie(uuidSecret, err) = GetSecretForInstance(instanceIdent);
+    if (err.IsNone()) {
+        return {uuidSecret, err};
+    }
+
+    uuidSecret = GenerateSecret();
+
+    err = AddSecret(uuidSecret, instanceIdent, instancePermissions);
+    if (!err.IsNone()) {
+        return {"", err};
+    }
+
+    return {uuidSecret, err};
+}
+
+Error PermHandler::UnregisterInstance(const InstanceIdent& instanceIdent)
+{
+    LockGuard lock(mMutex);
+
+    LOG_DBG() << "Unregister instance: instance = " << instanceIdent;
+
+    auto result = FindByInstanceIdent(instanceIdent);
+    if (!result.mError.IsNone()) {
+        LOG_WRN() << "Unregister instance not registered: instance = " << instanceIdent;
+
+        return result.mError;
+    }
+
+    return mInstancesPerms.Remove(result.mValue).mError;
+}
+
+Error PermHandler::GetPermissions(InstanceIdent& instanceIdent, Array<PermKeyValue>& servicePermissions,
+    const String& secretUUID, const String& funcServerID)
+{
+    LockGuard lock(mMutex);
+
+    LOG_DBG() << "Get permission: secretUUID = " << secretUUID << ", funcServerID = " << funcServerID;
+
+    const auto result = FindBySecretUUID(secretUUID);
+    if (!result.mError.IsNone()) {
+        return AOS_ERROR_WRAP(result.mError);
+    }
+
+    instanceIdent = result.mValue->mInstanceIdent;
+
+    for (const auto& it : result.mValue->mFuncServicePerms) {
+        if (it.mName == funcServerID) {
+            if (it.mPermissions.Size() > servicePermissions.MaxSize()) {
+                return AOS_ERROR_WRAP(ErrorEnum::eNoMemory);
+            }
+
+            servicePermissions = it.mPermissions;
+
+            return ErrorEnum::eNone;
+        }
+    }
+
+    return AOS_ERROR_WRAP(ErrorEnum::eNotFound);
```

Comment:
in some methods you are using AOS_ERROR_WRAP in some not, I think should be consistent

Reply:
Applied your suggestion